nil/internal/collate/syncer.go

```diff
@@ -413,22 +424,18 @@ func validateRepliedBlock(
 }
 
 func (s *Syncer) replayBlock(ctx context.Context, block *types.BlockWithExtractedData) error {
-	mainShardHash := block.Block.MainChainHash
-	if s.config.ShardId.IsMainShard() {
-		mainShardHash = block.Block.PrevBlock
+	roTx, err := s.db.CreateRoTx(ctx)
```

Comment:
In this and other places we create RO transactions in place. This is probably not a good thing to do as ideally we should create one top-level read-only stransaction instead. I don't think we should handle it in this ticket, but as a follow-up instead.